nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/avro/AvroRecordReader.java

```diff
@@ -33,14 +35,18 @@
 
     @Override
     public Record nextRecord(final boolean coerceTypes, final boolean dropUnknownFields) throws IOException, MalformedRecordException {
-        GenericRecord record = nextAvroRecord();
-        if (record == null) {
-            return null;
+        try {
+            GenericRecord record = nextAvroRecord();
+            if (record == null) {
+                return null;
+            }
+
+            final RecordSchema schema = getSchema();
+            final Map<String, Object> values = AvroTypeUtil.convertAvroRecordToMap(record, schema);
+            return new MapRecord(schema, values);
+        } catch (RuntimeException e) {
```

Comment:
Should we catch and rethrow the exceptions from the signature, then catch Throwable, to cover all our bases?

Reply:
indeed! 
However I was only able to add the catch clauses here in AvroRecordReader and not into CSVRecordReader: since the compiler checks the try {} 's body, and finds that it never throws those checked exceptions.

Reply:
Fair enough, but you can apply the `catch(Throwable)` to all the readers, just don't need a catch/rethrow for checked exceptions that aren't thrown. Seems a bit more future-proof than catching the RuntimeException we just found out about, in the future who knows what libraries will throw? :)

Reply:
indeed! sorry for the misunderstanding!
change is pushed now